### PR TITLE
Add support for `@removeTagHelpers` inheritance from _ViewStart files

### LINKED ISSUE #1657: Add support for `@removeTagHelpers` inheritance from _ViewStart files
We support `@addTagHelpers` being inherited from _ViewStart. `@removeTagHelpers` need to be similarly supported

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #1657

### CODE REVIEW

**Comment by pranavkm** [comment]
Updated PR.

File: src/Microsoft.AspNet.Mvc.Razor.Host/Directives/ChunkInheritanceUtility.cs [hunk 1]
```diff
@@ -39,14 +40,16 @@ public class ChunkInheritanceUtility
...
+        /// appears first.
         /// </summary>
         /// <param name="pagePath">The path of the page to locate inherited chunks for.</param>
-        /// <returns>A <see cref="IReadOnlyList{T}"/> of <see cref="Chunk"/> from _ViewStart pages.</returns>
-        public IReadOnlyList<Chunk> GetInheritedChunks([NotNull] string pagePath)
+        /// <returns>A <see cref="IReadOnlyList{CodeTree}"/> of parsed _ViewStart <see cref="CodeTree"/>s.</returns>
+        public IReadOnlyList<CodeTree> GetInheritedCodeTrees([NotNull] string pagePath)
         {
-            var inheritedChunks = new List<Chunk>();
+            var inheritedChunks = new List<CodeTree>();
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Mvc.Razor.Host/Directives/ChunkInheritanceUtility.cs]: `inheritedCodeTrees`

File: src/Microsoft.AspNet.Mvc.Razor.Host/Directives/ChunkInheritanceUtility.cs [hunk 2]
```diff
@@ -104,7 +108,12 @@ public IReadOnlyList<Chunk> GetInheritedChunks([NotNull] string pagePath)
             // In the second phase we invoke IChunkMerger.Merge for each chunk that has a mapped merger.
             // During this phase, the merger can either add to the CodeTree or ignore the chunk based on the merging
             // rules.
-            foreach (var chunk in inherited)
+            // Read the chunks outside in - that is chunks from the _ViewStart closest to the page get merged in first
+            // and the furthest one last. This allows the merger to ignore a directive like @model that was previously
+            // seen.
+            var chunksToMerge = inheritedCodeTrees.SelectMany(tree => tree.Chunks)
+                                               .Concat(_defaultInheritedChunks);
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Mvc.Razor.Host/Directives/ChunkInheritanceUtility.cs]: nit: super odd spacing :smile:

File: src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs [hunk 3]
```diff
@@ -183,8 +183,8 @@ public GeneratorResults GenerateCode(string rootRelativePath, Stream inputStream
         /// <inheritdoc />
         public override RazorParser DecorateRazorParser([NotNull] RazorParser razorParser, string sourceFileName)
         {
-            var inheritedChunks = ChunkInheritanceUtility.GetInheritedChunks(sourceFileName);
-            return new MvcRazorParser(razorParser, inheritedChunks);
+            var inheritedChunks = ChunkInheritanceUtility.GetInheritedCodeTrees(sourceFileName);
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs]: `inheritedCodeTrees`

**Comment by NTaylorMullen** [comment]
:shipit:

**Pull request closed without merging**